watertap/unit_models/tests/test_translator_adm1_asm1.py

```diff
@@ -0,0 +1,303 @@
+        assert hasattr(asmadm.fs.unit.outlet, "alkalinity")
+
+        assert number_variables(asmadm) == 129
+        assert number_total_constraints(asmadm) == 16
+
+        assert number_unused_variables(asmadm.fs.unit) == 4
+
+    @pytest.mark.component
+    def test_units(self, asmadm):
+        assert_units_consistent(asmadm)
+
+    @pytest.mark.unit
+    def test_dof(self, asmadm):
+        assert degrees_of_freedom(asmadm) == 0
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_initialize(self, asmadm):
+        initialization_tester(asmadm, optarg={"bound_push": 1e-8})
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solve(self, asmadm):
+        solver = get_solver(options={"bound_push": 1e-8})
+        results = solver.solve(asmadm)
+        assert_optimal_termination(results)
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solution(self, asmadm):
+        assert pytest.approx(101325.0, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.pressure[0]
+        )
+        assert pytest.approx(308.15, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.temperature[0]
+        )
+        assert pytest.approx(0.1308, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.conc_mass_comp[0, "S_I"]
+        )
+        assert pytest.approx(0.2585, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.conc_mass_comp[0, "S_S"]
+        )
+        assert pytest.approx(17.216, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.conc_mass_comp[0, "X_I"]
+        )
+        assert pytest.approx(3.2375, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.conc_mass_comp[0, "X_S"]
+        )
+        assert pytest.approx(1e-6, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.conc_mass_comp[0, "X_BH"]
+        )
+        assert pytest.approx(1e-6, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.conc_mass_comp[0, "X_BA"]
+        )
+        assert pytest.approx(1e-6, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.conc_mass_comp[0, "X_P"]
+        )
+        assert pytest.approx(1e-6, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.conc_mass_comp[0, "S_O"]
+        )
+        assert pytest.approx(1e-6, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.conc_mass_comp[0, "S_NO"]
+        )
+        assert pytest.approx(1.322, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.conc_mass_comp[0, "S_NH"]
+        )
+        assert pytest.approx(0.008, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.conc_mass_comp[0, "S_ND"]
+        )
+        assert pytest.approx(0.251, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.conc_mass_comp[0, "X_ND"]
+        )
+        assert pytest.approx(0.095061, abs=1e-2) == value(
```

Comment:
Shouldn't also check the `zero_flow_components`?

Reply:
We are testing them ("X_BH, "X_BA”, "X_P, "S_O, "S_NO")